dotnet/samples/Demos/Hosting/Agent/ChatWithAgent.ApiService/Program.cs

```diff
@@ -38,4 +44,30 @@ public static void Main(string[] args)
 
         app.Run();
     }
+
+    private static void AddKernelAndServices(WebApplicationBuilder builder)
+    {
+        // Add AzureOpenAI client.
+        builder.AddAzureOpenAIClient(
+            "azureOpenAI",
+            (settings) => settings.Credential = builder.Environment.IsProduction()
+                ? new DefaultAzureCredential()
```

Comment:
Should we also have key based example? Does the `DefaultAzureCredential()` includes the `ManagedIdentity` credential ?

```suggestion
                ? new DefaultAzureCredential()
```

Reply:
1. The key based example may be added in the next PR that will deal with reading model/agent parameters from config.
2. Yes, it does include the managed identity credential:
    <img width="640" alt="image" src="https://github.com/user-attachments/assets/70583392-ad02-4228-94b0-15c20be11e7e" />
Source: https://learn.microsoft.com/en-us/dotnet/azure/sdk/authentication/credential-chains?tabs=dac#defaultazurecredential-overview